src/app/logs/duck/sagas.ts

```diff
@@ -1,196 +1,130 @@
-import { select, call, put, take } from 'redux-saga/effects';
+import { select, call, put, take, takeEvery, all } from 'redux-saga/effects';
 import { ClientFactory } from '../../../client/client_factory';
 import { IClusterClient } from '../../../client/client';
 import { flatten } from 'lodash';
 import { LogActions, LogActionTypes } from './actions';
-import {
-  MigResource,
-  ExtendedCoreNamespacedResource,
-  CoreNamespacedResourceKind,
-  ExtendedCoreNamespacedResourceKind,
-  CoreNamespacedResource,
-  MigResourceKind
-} from '../../../client/resources';
-import { IMigPlan, IMigMigration } from '../../../client/resources/conversions';
 import Q from 'q';
-
-export enum LogKind {
-  velero = 'velero',
-  restic = 'restic',
-  controller = 'controller'
-}
-
-export enum ClusterKind {
-  source = 'source',
-  target = 'target',
-  host = 'host',
-}
-
-export interface ILog {
-  podName: string;
-  log: string;
+import { IDiscoveryClient } from '../../../client/discoveryClient';
+import { PlanPodReportDiscovery } from '../../../client/resources/discovery';
+import { IPlanReport, IPodLogSource, IPlanLogSources, ClusterKind } from '../../../client/resources/convension';
+import JSZip from 'jszip';
+import utils from '../../common/duck/utils';
+import { handleCertError } from './utils';
+import { AlertActions } from '../../common/duck/actions';
+
+function* downloadLog(action) {
+  const state = yield select();
+  const discoveryClient: IDiscoveryClient = ClientFactory.discovery(state);
+  const logPath: string = action.logPath;
+  try {
+    const archive = new JSZip();
+    const log = yield discoveryClient.getRaw(logPath);
+    const clusterName = logPath[4];
+    const logName = logPath[8];
+    archive.file(`${clusterName}-${logName}.log`, log.data.join('\n'));
+    const content = yield archive.generateAsync({ type: 'blob' });
+    const file = new Blob([content], { type: 'application/zip' });
+    const url = URL.createObjectURL(file);
+    yield put(LogActions.createLogArchive(url));
+  } catch (err) {
+    yield put(AlertActions.alertErrorTimeout(err.message));
+  }
 }
 
-export type IMigrationClusterLogType = {
-  [key in LogKind]?: ILog[];
-};
+function* downloadLogs(action) {
+  const state = yield select();
+  const discoveryClient: IDiscoveryClient = ClientFactory.discovery(state);
+  const report: IPlanLogSources = action.report;
+  try {
+    const archive = new JSZip();
+    const logPaths = flatten(Object.values(ClusterKind).map(
+      src => report[src].map(pod => pod.log)));
+
+    const logs = [];
+    for (const log of logPaths) {
```

Comment:
Necessary evil to do it sequentially, as concurrent requests broke the discovery a bit @jortel